commands/SummarizeCommand.ts

```diff
@@ -1,149 +1,89 @@
-                },
-            ],
-            temperature: 0,
-        };
-
-        const response = await http.post(OPENAI_URL, {
-            headers: {
-                Authorization: `Bearer ${OPENAI_API_KEY}`,
-                "Content-Type": "application/json",
-            },
-            content: JSON.stringify(body),
-        });
-
-        if (!response.content) {
-            await this.notifyMessage(
-                room,
-                read,
-                user,
-                "Something is wrong with AI. Please try again later"
-            );
-            throw new Error(
-                "Something is wrong with AI. Please try again later"
-            );
-        }
-
-        return JSON.parse(response.content).choices[0].message.content;
-    }
-
-    private async getThreadMessages(
-        room: IRoom,
-        read: IRead,
-        user: IUser,
-        threadId: string
-    ) {
-        const threadReader = read.getThreadReader();
-        const thread = await threadReader.getThreadById(threadId);
-
-        if (!thread) {
-            await this.notifyMessage(room, read, user, "Thread not found");
-            throw new Error("Thread not found");
-        }
-
-        const messageTexts: string[] = [];
-        for (const message of thread) {
-            if (message.text) {
-                messageTexts.push(`${message.sender.name}: ${message.text}`);
-            }
-        }
-
-        // threadReader repeats the first message once, so here we remove it
-        messageTexts.shift();
-        return messageTexts.join(" // ");
-    }
-
-    private async notifyMessage(
-        room: IRoom,
-        read: IRead,
-        user: IUser,
-        message: string,
-        threadId?: string
-    ): Promise<void> {
-        const notifier = read.getNotifier();
-
-        const messageBuilder = notifier.getMessageBuilder();
-        messageBuilder.setText(message);
-        messageBuilder.setRoom(room);
-
-        if (threadId) {
-            messageBuilder.setThreadId(threadId);
-        }
-
-        return notifier.notifyUser(user, messageBuilder.getMessage());
-    }
+	public command = 'ai-programmer';
+	public i18nParamsExample = 'Automatically create short codes according to specification';
+	public i18nDescription = '';
+	public providesPreview = false;
+	private readonly app: App;
+
+	constructor(app: App) {
+		this.app = app;
+	}
+
+	public async executor(
+		context: SlashCommandContext,
+		read: IRead,
+		modify: IModify,
+		http: IHttp
+	): Promise<void> {
+		const user = context.getSender();
+		const room = context.getRoom();
+		const threadId = context.getThreadId();
+
+		if (!threadId) {
+			await notifyMessage(
+				room,
+				read,
+				user,
+				'You can only call /ai-programmer in a thread'
+			);
+			throw new Error('You can only call /ai-programmer in a thread');
+		}
+
+		const messages = await this.getThreadMessages(room, read, user, threadId);
```

Comment:
@RyanbowZ  why is this check necessary ? We should be able to call the bot  in the channels input box.

Reply:
The current approach makes using the app a two-step process. Where someone would have to make a thread first, add info there and then use this slash command. This can be one of the features of the app ~ turn a thread into code, but cannot be the only way to use the app.